client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/TransferInvestigationsDialogs/TransferInvestigationDesk.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import { Autocomplete } from '@material-ui/lab';
+import { TextField, Typography } from '@material-ui/core';
+import { FormProvider, Controller, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers';
+
+import Desk from 'models/Desk';
+
+import useStyles from './TransferDialogsStyles';
+import validationSchema from './TransferInvestigationDeskSchema';
+import TransferInvestigationDialog from './TransferInvestigationDialog';
+import TransferInvestigationDeskInputsNames from './TransferInvestigationDeskInputsNames';
+
+const defaultDesk : Desk = {id: -1, deskName: ''};
+const defaultValues = {
+    [TransferInvestigationDeskInputsNames.DESK]: null,
+    [TransferInvestigationDeskInputsNames.REASON]: ''
+}
+
+const tranferDeskFormName = 'transferDesk';
+
+const TransferInvestigationDesk = (props: Props) => {
+
+    const { allDesks, open, onClose, onConfirm } = props;
+
+    const [transferDesk, setTransferDesk] = useState<Desk>(defaultDesk);
+    const [transferReason, setTransferReason] = useState<string>();
+
+    const methods = useForm({
+        mode: 'all',
+        resolver: yupResolver(validationSchema),
+        defaultValues
+    })
+
+    const onDialogConfirm = () => (
+        methods.handleSubmit((data) => {
+            onConfirm(data[TransferInvestigationDeskInputsNames.DESK], data[TransferInvestigationDeskInputsNames.REASON] || '');
+        })()
+    );
+
+    const classes = useStyles();
+
+    const isDataValid = React.useMemo<boolean>(() => (
```

Comment:
I think you can use isValid api from react-hook-form instead of implement it yourself. I don't know if it will work but worth a try

Reply:
Didnt find docs about it, dont want to use handle submit